tls/s2n_connection.c

```diff
@@ -938,6 +955,21 @@ int s2n_connection_set_write_fd(struct s2n_connection *conn, int wfd)
     return 0;
 }
 
+int s2n_connection_get_write_fd(struct s2n_connection *conn, int *writefd)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(writeFd);
+
+    /* in case we do not use managed io, or in case write fd was never set we should return invalid fd */
+    if(conn->managed_io == 1 && conn->send_io_context) {
+        struct s2n_socket_write_io_context *peer_socket_ctx = conn->send_io_context;
+        *writeFd = peer_socket_ctx->fd;
+    }
+    else {
+        *writeFd = -1;
+    }
```

Comment:
```suggestion
    POSIX_ENSURE_REF(writefd);

    /* in case we do not use managed io, or in case write fd was never set we should return invalid fd */
    if(conn->managed_io == 1 && conn->send_io_context) {
        struct s2n_socket_write_io_context *peer_socket_ctx = conn->send_io_context;
        *writefd = peer_socket_ctx->fd;
    }
    else {
        *writefd = -1;
    }
```